Automattic/wp-calypso
# Use DataViews component to render Billing History list

## Proposed Changes

* This PR replaces the current Billing History List with one that utilizes WordPress's DataViews component.  By doing this we can achieve a greater style consistency with the Sites management screen as well as incorporate features such as sorting by various columns and reorganizing them.
* This began by duplicating the existing Billing History List into a new folder and then using a feature flag to set which component would load in which environment.  
* WPCOM design requested we keep the header/navigation for the purchases section but they wanted us to implement a full-width layout.
* The "actions" kabob menu contains the view receipt and email receipt actions from the original version.
* Pagination, filtering, sorting, searching, and column reorganization are supported.

**Before**
<img width="1402" alt="before" src="https://github.com/user-attachments/assets/c45f6a78-3548-494e-bf73-c9d5ed7f452e" />

**After**
<img width="1459" alt="after" src="https://github.com/user-attachments/assets/347c3a7f-d54c-49d9-ba15-aef425419a6d" />



## Why are these changes being made?
When customers have large numbers of receipts, it can be difficult to find a receipt.  We wanted to offer the ability to sort the columns, and filter by more props as well.  Switching to the DataViews component makes this easier because it has so much functionality built-in.

This PR is being developed in parallel with changes in the Purchases list to allow similar sorting, searching, and filtering options.  It's in response to a long-standing feature request in https://github.com/Automattic/wp-calypso/issues/86616.


## Testing Instructions

1. Check out this branch and load Calypso locally.
2. Visit http://calypso.localhost:3000/me/purchases/billing and you should see your purchases from WPCOM. 
3. Use the search input to search for domain names, products purchased, and order amounts to see if the search works.
4. Next to the search box is an icon that opens up the filters.  Try clicking on it and applying different filters.  You can combine them as well.
5. The date, app, type, and amount columns are all sortable. Try sorting them ascending/descending.
6. At the bottom of the list of products, you'll see the pagination.  Use it to move through the pages and go to specific pages too.
7. Lastly, click on the gear icon and use the options there to customize the layout (like moving columns).

## TODOs
- [X] add/modify tests
- [ ] accessbility check

## Follow-ups
- [ ] Update receipt view with slide-in panel
- [ ] Update VAT layout to use slide-in panel as well

## Pre-merge Checklist
